packages/FormElement/src/FormElement.js

```diff
@@ -0,0 +1,169 @@
+import Description from "./components/Description";
+import ErrorMessage from "./components/ErrorMessage";
+import Hint from "./components/Hint";
+import Label from "./components/Label";
+
+import formElementStyles from "./FormElement.styles";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+
+  /** ID for the child element. */
+  id: PropTypes.string,
+
+  /** Should be disabled or not, default is false. */
+  isDisabled: PropTypes.bool,
+
+  /** Should label and children be inline or not, default is false. */
+  isInline: PropTypes.bool,
+
+  /** Should label be hidden, default is false. Note: this is discouraged because of accessibility requirements. */
+  isLabelVisuallyHidden: PropTypes.bool,
+
+  /** Should show is optional text besides the label or not. */
+  isOptional: PropTypes.bool,
+
+  /** Should be read-only or not, default is false. */
+  isReadOnly: PropTypes.bool,
+
+  /** Should show is required text besides the label or not. */
+  isRequired: PropTypes.bool,
+
+  /** Label text of this field. */
+  label: PropTypes.string.isRequired,
+
+  /** Size of the label, child component, error, hint and description (font size, min-height, padding, etc). */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  id: null,
+  isDisabled: false,
+  isInline: false,
+  isLabelVisuallyHidden: false,
+  isOptional: false,
+  isReadOnly: false,
+  isRequired: false,
+  size: "medium",
+};
+
+function FormElement(props) {
+  const {
+    children,
+    id,
+    isDisabled,
+    isInline,
+    isLabelVisuallyHidden,
+    isOptional,
+    isReadOnly,
+    isRequired,
+    label,
+    size,
+  } = props;
+
+  const extratedChildren = extractChildren(children, [
+    "FormElement.Description",
+    "FormElement.Error",
+    "FormElement.Hint",
+  ]);
+  const ariaDescriptionId = uuid();
+  const hasError = !!extratedChildren["FormElement.Error"] && !!extratedChildren["FormElement.Error"].props.children;
+  const uniqueId = isNil(id) || id === "" ? uuid() : id;
+  let isFooterInserted = false;
+
+  function renderFooter() {
+    if (isFooterInserted) return;
+
+    isFooterInserted = true;
+
+    if (hasError) {
+      return extratedChildren["FormElement.Error"];
+    }
+
+    if (extratedChildren["FormElement.Description"]) {
+      return React.cloneElement(extratedChildren["FormElement.Description"], {
+        ariaDescriptionId,
+      });
+    }
+
+    return null;
+  }
+
+  return (
+    <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled}>
+      <Label
+        hint={extratedChildren["FormElement.Hint"]}
+        id={uniqueId}
+        isInline={isInline}
+        isVisuallyHidden={isLabelVisuallyHidden}
+        isOptional={isOptional}
+        isRequired={isRequired}
+        label={label}
+      />
+
+      {extratedChildren.children.map(child => {
+        if (React.isValidElement(child)) {
+          const extendedProps = isString(child.type)
```

Comment:
While this might work, for now, I fear that this might failed in the feature seem more like abusing react api than a proper way though I don't have a solution on how to solved differently. No action just pointing a possible scenario.

Reply:
> What about changing `<FormElement.Hint />` to `<FormElement.Info />`

Decided use ``<FormElement.Help />``